### PR TITLE
kola: Add FailFast flag to test registration

### PR DESCRIPTION
**Pull request by arithx**
Adds a new flag (`FailFast`) to test registration, when set to true any
sub-tests will be skipped if a previous sub-test has failed.

### CODE REVIEW

**Comment by arithx** [comment]
Works for both tests that perform `c.Run` inside of the test code as well as `NativeFuncs` which perform that under the hood.

**Comment by arithx** [comment]
Pause on review while I investigate parallel sub-tests a bit more to make sure there won't be any future potential edge cases that aren't covered by this.

File: kola/register/register.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ type Test struct {
 	Architectures    []string // whitelist of machine architectures supported -- defaults to all
 	Flags            []Flag   // special-case options for this test
 
+	// FailFast skips any sub-test that occurs after a sub-test has
+	// failed.
+	FailFast bool
```
Comment (bgilbert) [inline kola/register/register.go]: More descriptive name?

Comment (arithx) [reply kola/register/register.go]: Any ideas? This sort of functionality in other test runners (including [golang](https://go-review.googlesource.com/c/go/+/74450)) usually uses `FailFast` or some derivative.

Comment (bgilbert) [reply kola/register/register.go]: Ah, if it's standard terminology let's stick with it.
